lib/Doctrine/DBAL/Exception/FormatArray.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Exception;
+
+use function array_map;
+use function bin2hex;
+use function implode;
+use function is_resource;
+use function is_string;
+use function json_encode;
+use function preg_replace;
+use function sprintf;
+
+final class FormatArray
+{
+    /**
+     * Returns a human-readable representation of an array of parameters.
+     * This properly handles binary data by returning a hex representation.
+     *
+     * @param array<mixed, mixed> $array
+     */
+    public function __invoke(array $array) : string
+    {
+        return '[' . implode(', ', array_map(static function ($param) {
```

Comment:
I may have misspoken about printing the entire thing. It may be an overkill and cause issues (e.g. exposing passwords or memory leaks). What we need is a printable representation of a variable in the context of the exception, not a dump.

In most cases where a function accepts `mixed $param`, the type (including class name or resource type) is enough. Printing the actual value may not be needed since the exception will most likely be about the type, not the value.

Reply:
I will remove this from the `FormatVariable` class:

```
if (is_array($value)) {
    return (new FormatArray())->__invoke($value);
}
```